ghcide/src/Development/IDE/Plugin/CodeAction/ExactPrint.hs

```diff
@@ -298,3 +295,113 @@ unqalDP paren =
+hideSymbol _ (L _ (XImportDecl _)) =
+    error "cannot happen"
+
+extendHiding ::
+    String ->
+    LImportDecl GhcPs ->
+    Maybe (Located [LIE GhcPs]) ->
+    DynFlags ->
+    TransformT (Either String) (LImportDecl GhcPs)
+extendHiding symbol (L l idecls) mlies df = do
+    L l' lies <- case mlies of
+        Nothing -> flip L [] <$> uniqueSrcSpanT
+        Just pr -> pure pr
+    let hasSibling = not $ null lies
+    src <- uniqueSrcSpanT
+    top <- uniqueSrcSpanT
+    rdr <- liftParseAST df symbol
+    let lie = L src $ IEName rdr
+        x = L top $ IEVar noExtField lie
+        singleHide = L l' [x]
+    when (isNothing mlies) $ do
+        addSimpleAnnT
+            singleHide
+            dp00
+            [ (G AnnHiding, DP (0, 1))
+            , (G AnnOpenP, DP (0, 1))
+            , (G AnnCloseP, DP (0, 0))
+            ]
+    addSimpleAnnT x (DP (0, 0)) []
+    addSimpleAnnT rdr dp00 $ unqalDP $ isOperator $ unLoc rdr
+    if hasSibling
+        then when hasSibling $ do
+            addTrailingCommaT x
+            addSimpleAnnT (head lies) (DP (0, 1)) []
+            unless (null $ tail lies) $
+                addTrailingCommaT (head lies) -- Why we need this?
+        else forM_ mlies $ \lies0 -> do
+            transferAnn lies0 singleHide id
+    return $ L l idecls {ideclHiding = Just (True, L l' $ x : lies)}
+    where
+        isOperator = not . all isAlphaNum . occNameString . rdrNameOcc
+
+deleteFromImport ::
+    String ->
+    LImportDecl GhcPs ->
+    Located [LIE GhcPs] ->
+    DynFlags ->
+    TransformT (Either String) (LImportDecl GhcPs)
+deleteFromImport (T.pack -> symbol) (L l idecl) llies@(L lieLoc lies) _ =do
+    let edited = L lieLoc deletedLies
+        lidecl' = L l $ idecl
+            { ideclHiding = Just (False, edited)
+            }
+    when (not (null lies) && null deletedLies) $ do
+        transferAnn llies edited id
+        addSimpleAnnT edited dp00
+            [(G AnnOpenP, DP (0, 1))
+            ,(G AnnCloseP, DP (0,0))
+            ]
+    pure lidecl'
+    where
+        deletedLies =
+            mapMaybe killLie lies
+        killLie :: LIE GhcPs -> Maybe (LIE GhcPs)
+        killLie v@(L _ (IEVar _ (L _ (unqualIEWrapName -> nam))))
+            | nam == symbol  = Nothing
+            | otherwise = Just v
+        killLie v@(L _ (IEThingAbs _ (L _ (unqualIEWrapName -> nam))))
+            | nam == symbol = Nothing
+            | otherwise = Just v
+
+        killLie (L lieL (IEThingWith xt ty@(L _ (unqualIEWrapName -> nam)) wild cons flds))
+            | nam == symbol = Nothing
+            | otherwise = Just $
+                L lieL $ IEThingWith xt ty wild
+                  (filter ((/= symbol) . unqualIEWrapName . unLoc) cons)
+                  (filter ((/= symbol) . T.pack . unpackFS . flLabel . unLoc) flds)
+        killLie v = Just v
+
+hideImplicitPreludeSymbol
+    :: String -> ParsedSource -> Rewrite
+hideImplicitPreludeSymbol symbol (L _ HsModule{..}) =
```

Comment:
I think it would be better to make this a total function.

Reply:
Sorry, I couldn't understand the suggestion. Isn't it already total? GHC didn't show any warning at least on my env.

Reply:
Ah, I mean is it possible to avoid using partial functions from Prelude, e.g. `last` or `fromJust`?

Reply:
Ah, I got it. I think the use of `last` is safe under the assumption that there must be at least one import decl already. As for `fromJust`... I'm not sure. If the module is an unsaved virtual one, it could be `Nothing`? Or such cases are rejected by the absence of `NormalizedFilePath`?

Reply:
OK, I've made `hideImplicitPreludeSymbol` total by returning `Maybe` value instead and ignores Prelude rewrite if it returns Nothing. Thank you for your suggestion!